github4s/shared/src/main/scala/github4s/api/Repos.scala

```diff
@@ -100,6 +100,33 @@ class Repos[C, M[_]](
       pagination = pagination
     )
 
+  /**
+    * List repositories that the authenticated user has explicit permission to access
+    *
+    * @param accessToken to identify the authenticated user
+    * @param headers optional user headers to include in the request
+    * @param visiblility can be all, public, private
+    * @param affiliation comma-separated list of owner, collaborator or organization members
+    * @param `type` visibility of the retrieved repositories, can be "all", "public", "private",
+    * "forks", "sources" or "member"
+    * @return GHResponse[List[Repository]] the list of repositories for this user
+    */
+  def listAuthenticatedUserRepos(
+     accessToken: Option[String] = None,
+     headers: Map[String, String] = Map(),
+     visibility: Option[String] = None,
```

Comment:
`visibility` and `affiliation` aren't being used within this request. Add them to the request query params like in `listCommits` endpoint. Keep in mind that you can't mix them with `type` param or github would return a 422 error.